### ISSUE TITLE
iOS Canvas memory exceeds the maximum limit when moving colorbar thresholds

### ISSUE THREAD

**Issue by Farkal** [nasa-gibs/worldview#1510]
**Describe the bug**
If you move the colorbar thresholds of a layer on an iOS device you will create lot of canvas and raise the maximum limit of the device (limit that just change with the new webkit release -> https://github.com/WebKit/webkit/commit/5d5b478917c685e50d1032ccf761ca53fc8f1b74#diff-b411cd4839e4bbc17b00570536abfa8f )

**To Reproduce**
1. Open the website
2. Add Cloud Top Height (Day)
3. Open Settings of the layer
4. Move threshold
5. Open safari debug console
6. Go to the Canvas tab
7. Move threshold
8. See the canvas memory use exceeds the maximum limit error (if you have lot of ram it can take some time but on iPad it come fast (the limit is 256MB)
9. From this moment safari don't want to allow you to create more canvas, openlayers loose references everything crash

**Expected behavior**
No error message and no crash

**Desktop (please complete the following information):**
 - OS: Mac, IPad, 
 - Browser safari
 - Version 12

**Additional context**
The problem comes from https://github.com/nasa-gibs/worldview/blob/360c06322c909164ec1fca117c896c1a053cf700/web/js/ol/lookupimagetile.js
Each time the colormap change a new canvas is created and also a new context. 
I use the same function to filter my data with openlayers because the `postcompose` don't work for me so if you find a way to resolve this i would be very grateful !

**Comment by mike-mcgann** [comment]
Thanks for the report. I'll take a look at it next week.

**Comment by mike-mcgann** [comment]
I am able to reproduce this bug.

**Comment by jasontk19** [comment]
Tested this with an iPad Air 2 with iOS 13.3.1 and did not see the app crash in either Safari or Chrome with multiple layers visible that had adjusted palette thresholds.  The app did become noticeably sluggish when adjusting thresholds though so I suspect the same issue exists but newer devices are able to keep up without crashing to a certain point (i.e. the iPad Air 2 has 2GB RAM).

**Comment by Farkal** [comment]
Yeah that's it, the more machine have RAM the more time you have before crash. But it still happen :/

**Comment by jasontk19** [comment]
@nasa-gibs/worldview This may be worth fixing as this could be a noticeable memory consumption issue on desktop machines as well.  Panning around the app for a few minutes with a custom palette (adjusted thresholds) enabled caused my Chrome tab to fairly quickly reach 1GB of usages while similar usage without custom palettes left it hovering pretty stably around ~470MB.

If a user had custom palettes enabled and left their tab open, continually using the app or task switching back to it throughout the day, this could conceivably cause their tab to freeze or crash eventually.

**Comment by hesam-oxe** [comment]
I see this has been open since 2018 with multiple engineers looking at it.

The root cause is clear from @Farkal's original analysis — 
`lookupimagetile.js` creates a new Canvas + context on every colormap 
change without destroying the previous one. 

The fix should be straightforward: either reuse the existing Canvas 
or explicitly destroy it before creating a new one. 

I can have a PR up shortly.
 /claim

**Comment by kartiksharmamrt1** [comment]
Noted looking into it

**Comment by minniewong** [comment]
This issue has been fixed with https://github.com/nasa-gibs/worldview/pull/6829
